File: BE/ecommerceproject/ecommerceproject/src/main/java/com/example/ecommerceproject/response/ApiResponse.java
package com.example.ecommerceproject.response;

import lombok.*;

@Getter
@Setter
@Data
@NoArgsConstructor
//@AllArgsConstructor
public class ApiResponse<T> {
    private int status;
    private String message;
    private T data;

    public ApiResponse(int status, String message) {
        this.status = status;
        this.message = message;
        this.data = null;
    }

    public ApiResponse(int status, String message, T data) {
        this.status = status;
        this.message = message;
        this.data = data;
    }
}

File: source/frontend-user/lib/core/services/api_response.dart
class ApiResponse<T> {
  final bool status;
  final String message;
  final T? data;

  ApiResponse({
    required this.status,
    required this.message,
    this.data,
  });
  
  @override
  String toString() {
    return 'ApiResponse{status: $status, message: $message, hasData: ${data != null}}';
  }
} 